### DISCUSSION TITLE
How to add border to lazy term?

### DISCUSSION THREAD

**Discussion by rollysys** [LazyVim/LazyVim#1915] [Q&A] [answered]
So sorry for bothered!
I'm new to neovim and lazyvim, and this is really awesome that I have uninstalled my vscode...
But I need help for "adding border to lazy term", and doesn't modify lazyvim under ~/.local/share/nvim folder.
I tried modifications below, but doesn't work
![image](https://github.com/LazyVim/LazyVim/assets/72813698/340c2485-e8d3-4d6d-81ec-2bcf7129dd4e)

**Comment by dpetka2001** [comment]
You can add a custom keymap in your `/lua/config/keymaps.lua` like so
```lua
vim.keymap.set("n", "<C-/>", function()
  Util.terminal(nil, { border = "rounded" })
end, { desc = "Term with border" })
```
You can see what values the border can get [here](https://github.com/folke/lazy.nvim/blob/16603c6917435d8446f7357cb61095138a417085/lua/lazy/view/float.lua#L12) in the `lazy.nvim` code.

**Reply by TalpsG** [reply]
i just add your code in keymaps.lua and it works normally but doesnt work in tmux. is there any setups for tmux?  
in wezterm 
![2024-02-24T21:31:22,146862072+08:00](https://github.com/LazyVim/LazyVim/assets/135484898/f2d3d3fb-9306-4e47-9237-317f8bdc902e)

in tmux 
![2024-02-24T21:32:51,370301448+08:00](https://github.com/LazyVim/LazyVim/assets/135484898/a9746c71-5878-4bad-a5a2-2cf4e6745441)

**Reply by dpetka2001** [reply]
`tmux` interprets `<C-/>` as `<C-_>` probably, so just change your keymap.

**Reply by TalpsG** [reply]
it works ,thank you !!!

**Comment by rollysys** [comment]
it doesn't work...
![image](https://github.com/LazyVim/LazyVim/assets/72813698/de1f0a7e-c7b1-4852-9e0d-2acc6aac3be2)
![image](https://github.com/LazyVim/LazyVim/assets/72813698/9783e3c4-6f83-45c9-a48d-fb721f324fd4)

**Reply by dpetka2001** [reply]
Do you have `local Util = require("lazyvim.util")` at the top of your `keymaps.lua` file?

**Reply by rollysys** [reply]
Yes, I did, but still not works.
Btw, if I change <C-/> to <C-n>, the floating terminal will have border now, but can't disappear 
So it seems like the default <C-/> not be overwritten?
Thanks

**Reply by dpetka2001** [reply]
For me the code that I posted works just fine. There must be something else with your config. I tested it with a new Lazyvim starter template.
Try the following: `:<C-v><C-/>` what does it show?

**Reply by rollysys** [reply]
I have tried a clean Lazyvim starter, and below  is the `git diff` output in the ~/.config/nvim_clean folder:
![image](https://github.com/LazyVim/LazyVim/assets/72813698/4799e921-39a8-4de4-95ac-c0ee2dac164f)
but the terminal still not have border
Btw, I'm in wsl2 ubuntu 22.04 of windows 10

**Reply by dpetka2001** [reply]
Try the following: `:<C-v><C-/>` what does it show?

**Reply by rollysys** [reply]
![image](https://github.com/LazyVim/LazyVim/assets/72813698/dcfa6315-9e2e-4f55-b7b3-f700ff8c02d0)

**Reply by dpetka2001** [reply]
No, I meant open command prompt by pressing `:` and then press `<C-v><C-/>` in the command prompt of Neovim.

**Reply by rollysys** [reply]
![image](https://github.com/LazyVim/LazyVim/assets/72813698/ca6c747e-6ce0-412a-b588-fb07e5d48432)

**Reply by dpetka2001** [reply]
Ok, it's a problem with how your terminal interprets keycodes. In your keymap, instead of `<C-/>` use `<C-_>` and try if it works.

**Reply by rollysys** [reply]
It works now, thanks so much!!
But why? can you kindly give an explanation?
Thanks!

**Reply by dpetka2001** [reply, marked as answer]
It's because of how your terminal interprets the keycodes from the keys being pressed. It is quite frequent, from what I've read and googled, that some terminals interpret `<C-/>` as `<C-_>`. I don't know the technical details behind this, but this is also why Lazyvim has 2 [mappings](https://github.com/LazyVim/LazyVim/blob/e5babf289c5ccd91bcd068bfc623335eb76cbc1f/lua/lazyvim/config/keymaps.lua#L144-L145), so that most terminals would work. In your case, since your terminal interprets `<C-/>` as `<C-_>`, we had to change the 2nd mapping to change it to do what you want. For a more detailed answer, you can also read through this [post](https://apple.stackexchange.com/questions/24261/how-do-i-send-c-that-is-control-slash-to-the-terminal) which explains things better than I can.

**Reply by rollysys** [reply]
Thanks so much!
